### PR TITLE
`azurerm_app_service_managed_certificate` - create certificate in service plan resource group

### LINKED ISSUE #9699: azurerm_app_service_managed_certificate stuck in create loop when app and service plan are in different resource groups
### Versions

Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/azurerm v2.39.0

### Affected Resource

* `azurerm_app_service_managed_certificate`

### Problem

Resource creation is stuck waiting for resource creation even though the resource is already created. This happens when the web app and the service plan are in different resource groups. This is because the certificate is force created in the app service plan resource group by ARM even if the resource group specified is that of the app.

This needs to be fixed by looking up the resource group of the app service plan, and specifying that as the resource group of the certificate explicitly. If this isn't don't, ARM does it implicitly anyway, which is why the created certificate resource isn't found.

### Terraform Configuration Files

```hcl
locals {
  location = "australiaeast"

  //existing dns_zone details
  dns_name = "example.com"
  dns_rg   = "mydnsrg"
  dns_sub  = "xxxxxxxxx-xxxxxx-xxxxxxx-xxxxxx"
}

resource "random_integer" "id" {
  min     = 1
  max     = 50000
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "app" {
  name     = "test-app"
  location = local.location
}

resource "azurerm_resource_group" "host" {
  name     = "test-host"
  location = local.location
}

resource "azurerm_app_service_plan" "example" {
  name                = "appserviceplan"
  location            = azurerm_resource_group.host.location
  resource_group_name = azurerm_resource_group.host.name

  sku {
    tier = "Premium"
    size = "P1"
  }
}

resource "azurerm_app_service" "example" {
  name                = "myspecialapp-${random_integer.id.result}"
  location            = azurerm_resource_group.app.location
  resource_group_name = azurerm_resource_group.app.name
  app_service_plan_id = azurerm_app_service_plan.example.id
}

// subscription for where the dns_zone is
provider "azurerm" {
  features {}
  alias                      = "dns"
  subscription_id            = local.dns_sub
  skip_provider_registration = true
}

data "azurerm_dns_zone" "main" {
  provider            = azurerm.dns
  name                = local.dns_name
  resource_group_name = local.dns_rg
}

resource "azurerm_dns_cname_record" "main" {
  provider            = azurerm.dns
  name                = "test"
  zone_name           = data.azurerm_dns_zone.main.name
  resource_group_name = data.azurerm_dns_zone.main.resource_group_name
  ttl                 = 300
  record              = azurerm_app_service.example.default_site_hostname
}

resource "azurerm_dns_txt_record" "main" {
  provider            = azurerm.dns
  name                = "asuid.${azurerm_dns_cname_record.main.name}"
  zone_name           = data.azurerm_dns_zone.main.name
  resource_group_name = data.azurerm_dns_zone.main.resource_group_name
  ttl                 = 300
  record {
    value = azurerm_app_service.example.custom_domain_verification_id
  }
}

resource "azurerm_app_service_custom_hostname_binding" "main" {
  hostname            = trim(azurerm_dns_cname_record.main.fqdn, ".")
  app_service_name    = azurerm_app_service.example.name
  resource_group_name = azurerm_resource_group.app.name
  depends_on          = [azurerm_dns_txt_record.main]
  lifecycle {
    ignore_changes = [ssl_state, thumbprint]
  }
}

resource "azurerm_app_service_managed_certificate" "main" {
  custom_hostname_binding_id = azurerm_app_service_custom_hostname_binding.main.id
}
```

### PR DESCRIPTION
**Pull request by AdamCoulterOz**
Fixes #9699 

FYI - @jackofallops

### CODE REVIEW

File: azurerm/internal/services/web/app_service_managed_certificate_resource.go [hunk 1]
```diff
@@ -117,14 +117,19 @@ func resourceArmAppServiceManagedCertificateCreateUpdate(d *schema.ResourceData,
 	}
 	appServicePlanID = *appService.SiteProperties.ServerFarmID
 
+	appServicePlanIDParts, err := parse.AppServicePlanID(appServicePlanID)
```
Comment (jackofallops) [inline azurerm/internal/services/web/app_service_managed_certificate_resource.go:120]: Really minor, but for the sake of convention, could you refactor the names of these vars for:
`appServicePlanID` => `appServicePlanIDRaw`
and
`appServicePlanIDParts` => `appServicePlanID`

**Review by jackofallops — CHANGES_REQUESTED** [review]
Thanks for the catch @AdamCoulterOz - one minor change requested, but otherwise will be good to merge.

File: azurerm/internal/services/web/app_service_managed_certificate_resource.go (hunk 1 above)
Comment (AdamCoulterOz) [reply azurerm/internal/services/web/app_service_managed_certificate_resource.go:120]: haha... I saw that when I was doing it last night... ill fix it now

**Comment by AdamCoulterOz** [comment]
@jackofallops just a reminder this is small fix is still waiting... we need this to make 2.40 also in order to make use of all the other work. I already addressed your previous feedback.

**Review by jackofallops — APPROVED** [review]
Thanks @AdamCoulterOz - LGTM now 👍

**Comment by jackofallops** [comment]
Tests passing:
```
=== RUN   TestAccAzureRMAppServiceManagedCertificate_basicLinux
=== PAUSE TestAccAzureRMAppServiceManagedCertificate_basicLinux
=== CONT  TestAccAzureRMAppServiceManagedCertificate_basicLinux
--- PASS: TestAccAzureRMAppServiceManagedCertificate_basicLinux (225.23s)
=== RUN   TestAccAzureRMAppServiceManagedCertificate_requiresImport
=== PAUSE TestAccAzureRMAppServiceManagedCertificate_requiresImport
=== CONT  TestAccAzureRMAppServiceManagedCertificate_requiresImport
--- PASS: TestAccAzureRMAppServiceManagedCertificate_requiresImport (229.70s)
=== RUN   TestAccAzureRMAppServiceManagedCertificate_basicWindows
=== PAUSE TestAccAzureRMAppServiceManagedCertificate_basicWindows
=== CONT  TestAccAzureRMAppServiceManagedCertificate_basicWindows
--- PASS: TestAccAzureRMAppServiceManagedCertificate_basicWindows (220.59s)
PASS
```

**Pull request merged**